### PR TITLE
Read the clearance register, and fall back to Haiku when no model answers

### PR DESCRIPTION
**Pull request by norrisaftcc**
## Changes Summary

Two independent pieces, stacked on #56.

### 1. Clearance is read from a register, never inferred (S1-09)

Nothing ever read a clearance source. `clearance_level` round-tripped the ledger and nothing else wrote it, so **every citizen was permanently RED** and the INFRARED and BLUE+ branches were built, tested, and unreachable.

The missing source is `.shodann/clearances.json` in the citizen's own repository — a flat username-to-band map, the shape Landmine 8 records as having no example anywhere. There is one now, and a test asserts it parses.

**The instructor sets it. SHODANN never infers it, at any band.** `design_docs/CLEARANCE_REGISTER.md` already made that argument for GREEN and BLUE+ on the grounds that those are role assignments rather than measurements; it generalises. A band inferred from readings is a second score, and this product rests on improvement outranking position. `prompts/03`'s `INFER_CLEARANCE` is therefore **declined rather than unimplemented** — and its documented defects (no terminal `ELSE`, an ORANGE branch shadowing YELLOW) are consistent with that. Nobody finished it because it should not exist. The register now says so.

- **The file outranks the ledger.** The ledger keeps the band so history stays readable, but an instructor who promotes a citizen must not wait for a merge to rewrite a stored value before the promotion means anything.
- **Everyone starts at RED.** INFRARED stays in the ladder because the register defines it and the renderer must not fail on it, but it is an onboarding state — a citizen without a GitHub account has no record to hold a band.
- **Disclosure waits for ORANGE.** Not secrecy: the file is in the citizen's own repository and readable from their first day. What waits is *being told*, because a beginner handed a knob controlling how much explanation they receive will reasonably turn it down before they know what they are turning down.
- Every failure reading the register is an unset band rather than an exception, and out-of-range values saturate. A citizen must not lose their review to a trailing comma in a config file.

The disclosure footer is **appended after validation** rather than requested from the model — a model asked to reproduce a file path will eventually reproduce a wrong one, and this way it never competes for the 400-word budget.

### 2. Anthropic Haiku as the fallback provider (rung 3c)

`llm.py` spoke one wire format and its docstring called that deliberate: OpenAI-compatible `/chat/completions` makes local inference a config change, because Gemini publishes such an endpoint and Ollama serves one natively.

**Anthropic does not.** Different request body, different response shape, `x-api-key` rather than a bearer token, and a `refusal` stop reason with no counterpart. So this is a port, and the honest way to write it is through the official SDK — which already handles the refusal, the typed errors, and the retries that hand-rolled urllib would get subtly wrong. The dependency buys deleting all of that rather than adding an abstraction layer.

- **The wire is explicit, never sniffed from the URL.** Inferring a protocol from a hostname means a deployment changes provider by editing something that looks like an address. A config pointed at `api.anthropic.com` on the OpenAI wire stays on the OpenAI wire, and a test asserts it.
- **Unreachability is the only trigger.** A response that violates the output contract twice raises `_ContractViolation`, which the chain re-raises rather than falling back on — the first model understood the prompt perfectly well, so a second provider is not the missing piece and falling back would spend a second bill to reach the same comment.
- **Only an environment-resolved primary gets the environment's fallback.** An explicit config means an explicit fallback or none — which is what keeps `scripts/dev.py render` offline on a machine with `ANTHROPIC_API_KEY` exported.
- No `thinking` parameter: Haiku 4.5 predates adaptive thinking, and job 4 reframes facts that deterministic tooling already established.

## One defect caught before it shipped

An unset GitHub Actions variable arrives as **the empty string, not as an absent key**. `env.get(name, DEFAULT)` therefore returns `""`, the model name disappears, and `configured` reports "no fallback set up" on a deployment that supplied a key. Every read now uses `or DEFAULT` (with a `.strip()`, since a whitespace-only value is truthy) and the timeout parses defensively.

That is the `EARLY_RUNS.md` shape exactly — green everywhere, wrong in production — and it would have presented as "the fallback silently never fires", which is the hardest kind to notice.

## Related Issues

None. The survey items live in `design_docs/sprints/2026-07-28/01-candidates.md`. Addresses **S1-09**.

## Component(s) Affected

- [x] Core Workflow (GitHub Actions)
- [ ] Velocity Calculation Engine
- [ ] RAGE STATE (Security Mode)
- [x] Persona/Voice System
- [x] State Management
- [ ] Templates
- [x] Documentation
- [x] Other: model transport (`src/shodann/llm.py`)

## Testing Performed

- [x] Manual testing completed
- [x] Tested with sample data/workflows
- [x] Reviewed output/behavior
- [x] Edge cases considered

**Testing Details:**

**314 passed, 3 skipped**, ruff clean (288 after the clearance half, 305 after the fallback, 314 with the contract test).

Each new guard was verified to fail against the defect it describes:

| Guard | Reverted to | Result |
|---|---|---|
| register reaches the review | override + disclosure removed from `review()` | 2 failed, 47 passed |
| fallback key mapped in CI | `ANTHROPIC_API_KEY` removed from the workflow | 1 failed, 12 passed |

Measured, not asserted:

- `ANTHROPIC_API_KEY=sk-would-be-billed python scripts/dev.py render` → still takes the REDUCED ALLOCATION path. The offline guarantee holds under a live key.
- A register naming a band the ledger disagrees with renders `**Clearance**: GREEN`, not the stored RED.
- A RED citizen's comment contains no reference to `.shodann/clearances.json`; an ORANGE citizen's does, positioned after the readings.
- A contract violation leaves the fake SDK client at `calls == 0` — the fallback is genuinely not reached.
- Malformed registers (trailing comma, wrong root type, non-numeric band, missing file) all resolve to *unset* without raising.

## Documentation Updated

- [x] Code comments added/updated
- [x] Design docs updated
- [ ] README updated
- [ ] User-facing documentation updated
- [ ] No documentation changes needed

**Files Updated:** `design_docs/CLEARANCE_REGISTER.md` gains a "Who sets the register" section recording the decision, the file shape, and the `INFER_CLEARANCE` decline. `CLAUDE.md` § Toolchain gains one line on the two wire formats. `pyproject.toml` explains why `anthropic` is a dependency and why it is deliberately unpinned — the freeze covers tools whose output feeds the score, and this one produces prose the validator and groundedness probe both re-check.

### Student-Facing Changes

**Two, both intended.**

A citizen's band can now actually differ from RED, which changes register, word budget, and section structure — those branches existed and were tested but no citizen could reach them. And from ORANGE up, every review carries a footer naming the file that governs how they are spoken to.

Nothing changes for a citizen at RED with no register present, which is every citizen today.

### Pedagogical Alignment

- [x] Maintains growth-positive framing
- [x] Appropriate for target clearance level(s)
- [x] Preserves SHODANN persona/voice consistency
- [x] Supports velocity-over-position principle
- [x] No negative impact on learning experience
- [ ] Not student-facing

Declining `INFER_CLEARANCE` is the velocity-over-position item: an inferred band is a second score. The ORANGE disclosure is the pedagogical one — the knob is disclosed when a citizen can evaluate what turning it down would cost them.

### Clearance Levels Affected

- [x] INFRARED (Complete beginner)
- [x] RED (Junior)
- [x] ORANGE (Mid-level)
- [x] YELLOW (Senior)
- [x] GREEN (Lead)
- [x] BLUE+ (Strategic)
- [ ] All levels
- [ ] Not student-facing

Every band, because until now only one of them was reachable.

## Pre-Merge Checklist

- [x] Code follows project style and conventions
- [x] Comments added for complex logic
- [x] Documentation updated (if needed)
- [x] Design docs updated if architecture changed
- [x] Persona voice maintained (if student-facing content) — the disclosure footer is asserted against the forbidden-vocabulary list
- [x] No hardcoded credentials or sensitive data — `describe()` and the dataclass `repr` are both asserted not to leak the key
- [x] Testing completed successfully
- [x] Related issues linked above — explicitly none, with reasoning
- [x] Educational impact assessed above

## Additional Notes

**The register ships with `norrisaftcc` at RED**, which is indistinguishable from the current default and therefore proves nothing on its own. Changing that `"2"` to `"6"` is the one-character way to exercise the BLUE+ path live — different heading (`Observations` rather than `Recommended Iteration`), no "exactly 1 action" rule, peer register. That is a real change to your own reviews, so it is left as a deliberate act rather than made on your behalf.

A fallback-served review is **silent about being a fallback**. It is not a degraded outcome — the citizen got a complete review — so no footer is added. Which model served is useful maintainer telemetry and is not currently recorded anywhere; worth deciding separately.

---
_Generated by [Claude Code](https://claude.ai/code/session_01MDFNeETL6SFPY2PYfwyBiK)_

### CODE REVIEW

File: src/shodann/review.py [hunk 1]
```diff
@@ -462,13 +524,15 @@ def review(
...
         prompt = render_prompt(context)
-        body = _synthesise(prompt, spec, config, opener=opener)
+        body = _synthesise_chain(prompt, spec, config, fallback, opener=opener, client=client)
     except _AlreadyResolved:
         pass
     except LLMUnavailable as unavailable:
         degradation = str(unavailable)
         body = reduced_allocation_comment(facts, result, record, degradation, reports)
 
+    body += clearance_disclosure(record.clearance_level)
```
[the remark this answers is not in the document]
Comment (norrisaftcc) [reply src/shodann/review.py]: Correct, and worse than described — fixed in `3ed9ce0`, though not by dropping the footer.

**Why not drop it.** Disclosure would then arrive or not depending on how verbose the model happened to be that run, and a promotion the citizen is told about only sometimes is not a promotion. Instead the footer's cost is *reserved from the budget before the model is told what its budget is*: 370 words of review plus 30 of footer, so the posted comment stays inside the 400 the contract states. The cap is on the comment a citizen receives, and the footer is part of that comment.

**The part the diff doesn't show.** The degraded path was already over. `reduced_allocation_comment` is SHODANN's own text at a fixed ~235 words against a 250-word cap, so it can neither shrink to make room nor absorb a reservation — subtracting 30 would have put it over its own cap immediately. Its budget is now stated as 250 of review plus the 30-word reservation. An ORANGE citizen on that path would have received a comment failing the contract SHODANN enforces on everyone else.

Nothing caught it because every test citizen is RED. My first regression test didn't catch it either: built against an empty temporary repository, it produced a comment 37 words shorter than a real one — short enough to pass a cap the real thing breaks. It now measures against this repository and fails at bands 3 through 6 without the fix.

The other comment was right too and is fixed as suggested: `_synthesise_chain`'s docstring claimed unreachability was the only trigger, which is false — `generate()` raises `LLMUnavailable` for unparseable JSON, an unexpected shape, a refusal and an empty body alike. The behaviour is intended (a provider answering with garbage is no more available than one not answering), so the docstring was corrected rather than the code narrowed.

---
_Generated by [Claude Code](https://claude.ai/code)_
